src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -95,6 +95,16 @@ struct extent_to_write_t {
 };
 using extent_to_write_list_t = std::list<extent_to_write_t>;
 
+struct po_meta_t {
+  enum class po_type_t {
+    NONE, 
+    OVERWRITE,
+    REMAP
+  } type = po_type_t::NONE;
+  uint64_t laddr_start;
+  extent_len_t length;
+}; // partial-overwrite
```

Comment:
Hmm, instead of containing it inside the insert request `extent_to_insert_t`, how about introducing another type for delta-based overwrite in `extent_to_remap_t` ?

Specifically, rename `extent_to_remap_t::type_t::REMAP` to `REMAP1`, `OVERWRITE` to `REMAP2` (it is just a varient of `REMAP1` to remap 2 ranges), and reuse `OVERWRITE` for the delta-based overwrite request.

The reason behind is that delta-based overwrite is alternative to the current remap-based overwrite, so grouping them together is logically more reasonable.

Reply:
Done.